### PR TITLE
F/serve command for all plugin types

### PR DESCRIPTION
**Pull request by haoliangyu**
This PR updates the `serve` command so that it can support all plugin types. See the updated README for more details.

### CODE REVIEW

File: src/commands/serve.js [hunk 1]
```diff
@@ -1,48 +1,33 @@
...
+    .positional('path', {
+      description: 'server file path',
+      type: 'string'
+    })
     .option('port', {
       alias: 'p',
       type: 'number',
       description: 'port number of the server'
     })
+    .option('data', {
```
Comment (rgwozdz) [inline src/commands/serve.js]: Very cool option!

File: src/utils/serve/get-provider.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+const fs = require('fs-extra')
```
Comment (rgwozdz) [inline src/utils/serve/get-provider.js]: Awesome.

File: src/utils/serve/index.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    // needed and here we provide a simple GeoJSON provider
+    if (koopConfig.type !== 'provider') {
+      const dataPath = options.data
+
+      if (
+        !dataPath ||
+        !dataPath.endsWith('.geojson') ||
+        !(await fs.pathExists(path.join(cwd, dataPath)))
+      ) {
+        throw new Error('A GeoJSON test data is requried for the dev server.')
```
Comment (rgwozdz) [inline src/utils/serve/index.js]: I think you left out a word or two here.

File: test/utils/serve/index.test.js [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+    const appPath = path.join(temp, appName)
+    const serve = proxyquire(moduleName, {
+      '../exec-realtime': (command) => {
+        const expected = path.join(appPath, 'app.js')
+        expect(command).to.equal(`node ${expected}`)
+        done()
+      }
+    })
+
+    createNewProject(temp, 'app', appName, defaultOptions)
```
Comment (rgwozdz) [inline test/utils/serve/index.test.js]: How come promises instead of await?

File: test/utils/serve/index.test.js [hunk 5]
```diff
@@ -0,0 +1,150 @@
+/* eslint-env mocha */
```
Comment (rgwozdz) [inline test/utils/serve/index.test.js]: Great set of new tests!

File: test/utils/serve/index.test.js (hunk 4 above)
Comment (haoliangyu) [reply test/utils/serve/index.test.js]: The `serve()` function executes a shell command or starts a server. Both operations don't have an exit and cannot use `await`. I find the callback `done()` is more useful when mocking the dependencies.

But if I use the `done()` function and `async` function at the same time, Mocha complains about it and only allows to either return a promise or use the callback. So I have to use the old-style promise chain but not return the promise.

**Pull request merged**